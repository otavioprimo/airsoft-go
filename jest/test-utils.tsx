import React from 'react';
import {render} from '@testing-library/react-native';
import {ThemeProvider} from 'styled-components/native';
import {createStore, Reducer} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import * as Theme from '../src/theme';

interface ReduxProviders {
  state: any;
  reducer: Reducer<any, any>;
}

interface RenderProviders {
  children: React.ReactElement;
  options?: ReduxProviders;
}

const ReduxProvider = ({children, options}: any) => {
  if (!options) {
    return <>{children}</>;
  }

  const store = createStore(options.reducer, options.state);
  return <Provider store={store}>{children}</Provider>;
};

const AllTheProviders: React.FC<RenderProviders> = ({children, options}) => {
  return (
    <ThemeProvider theme={Theme.DarkTheme}>
      <ReduxProvider options={options}>{children}</ReduxProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: ReduxProviders) => {
  return render(<AllTheProviders options={options}>{ui}</AllTheProviders>);
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
