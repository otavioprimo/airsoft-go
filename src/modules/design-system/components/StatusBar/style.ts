import styled from 'styled-components/native';
import {getStatusBarHeight} from 'react-native-status-bar-height';

export const Container = styled.View`
  background-color: ${({theme}) => theme.primary};
  height: ${getStatusBarHeight()}px;
`;
