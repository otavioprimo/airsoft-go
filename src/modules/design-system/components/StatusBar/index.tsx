import React from 'react';
import {StatusBar as RNStatusBar, Platform} from 'react-native';
import {Container} from './style';

interface StatusBarProps {
  isDarkMode?: boolean;
}

const StatusBar: React.FC<StatusBarProps> = ({isDarkMode}) => {
  const content = () => {
    return (
      <RNStatusBar
        translucent
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor={isDarkMode ? '#000' : '#222'}
      />
    );
  };
  return Platform.select({
    ios: <Container>{content()}</Container>,
    default: content(),
  });
};

export default React.memo(StatusBar);
