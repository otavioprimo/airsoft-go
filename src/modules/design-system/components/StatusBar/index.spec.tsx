import React from 'react';
import {render} from 'test/test-utils';
import StatusBar from './index';

describe('Design System - StatusBar', () => {
  it('Should match snapshot in light mode', () => {
    const {toJSON} = render(<StatusBar isDarkMode={false} />);
    expect(toJSON()).toMatchSnapshot('DS_StatusBar_Light');
  });

  it('Should match snapshot in dark mode', () => {
    const {toJSON} = render(<StatusBar isDarkMode />);
    expect(toJSON()).toMatchSnapshot('DS_StatusBar_Dark');
  });
});
