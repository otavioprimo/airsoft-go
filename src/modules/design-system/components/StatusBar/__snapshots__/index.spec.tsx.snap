// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System - StatusBar Should match snapshot in dark mode: DS_StatusBar_Dark 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#222",
        "height": 20,
      },
    ]
  }
/>
`;

exports[`Design System - StatusBar Should match snapshot in light mode: DS_StatusBar_Light 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#222",
        "height": 20,
      },
    ]
  }
/>
`;
